crates/fuel-core/src/service/adapters/graphql_api.rs

```diff
@@ -117,6 +118,11 @@ impl DatabaseMessages for Database {
             .into_boxed()
     }
 }
+impl DatabaseSpentMessages for Database {
```

Comment:
We don't need an additional method. `owned_message_ids` and `all_messages` should return messages with this information.